R/method-correlation.R

```diff
@@ -111,8 +111,16 @@ function(
   }
 
   # Check that all values are numeric
-  if (!identical(veupathUtils::findNumericCols(data1), names(data1))) { stop("All columns in data1 must be numeric.")}
-  if (!identical(veupathUtils::findNumericCols(data2), names(data2))) { stop("All columns in data2 must be numeric.")}
+  if (!identical(veupathUtils::findNumericCols(data1), names(data1))) { 
+    warning("All columns in data1 are not numeric. Only numeric columns will be used.")
+    keepCols <- veupathUtils::findNumericCols(data1)
+    data1 <- data1[, ..keepCols]
+  }
+  if (!identical(veupathUtils::findNumericCols(data2), names(data2))) { 
+    warning("All columns in data2 are not numeric. Only numeric columns will be used.")
+    keepCols <- veupathUtils::findNumericCols(data2)
+    data2 <- data2[, ..keepCols]
+  }
```

Comment:
Do we need a check here now to ensure data1 and data2 are not empty? Or alternatively that length(keepCols) > 1?